Dockerfile

```
@@ -0,0 +1,44 @@
+ARG VOLTA_TAG=v2.0.1
+ARG DUFS_VERSION=0.43.0
+
+
+# build the app via volta to use the correct node version
+FROM rust:latest AS build-app
+RUN apt update -y && apt upgrade -y
+WORKDIR /app
+
+# install volta directly from source
+ARG VOLTA_TAG
+RUN cargo install --git https://github.com/volta-cli/volta.git --tag $VOLTA_TAG --locked
+ENV PATH="bin/volta/bin:$PATH"
+
+# install node
+COPY package.json .
```

Use `COPY --link` to allow caching the `package.json` file if a previous step might invalidate them.

https://docs.docker.com/reference/dockerfile/#benefits-of-using---link

```suggestion
COPY --link package.json .
```